### DISCUSSION TITLE
memtest86+ EFI

### DISCUSSION THREAD

**Discussion by returntrip** [netbootxyz/netboot.xyz#681] [General] [answered]
Is memtest86+ meant to work with the EFI version (netboot.xyz.efi)?

I can get it working only with the Legacy BIOS  bootloader (netboot.xyz.kpxe).

Thanks in advance.

**Comment by antonym** [comment]
From my understanding, I don't believe there is a free EFI PXE version of memtest86+ anymore.  Only the Pro version contains PXE support unfortunately:

https://www.memtest86.com/compare.html

**Reply by returntrip** [reply]
You are probably right, but then how is the BIOS PXE still working with netboot.xyz? According to the link you have posted PXE boot is available only in the Site edition.

BTW, love netboot.xyz. thanks for all the hard work!

**Reply by antonym** [reply]
It's an older version which I believe was open source, apparently there's an EFI PXE version that's floating around before they made it licensed but I have not been able to track it down yet.

**Reply by EnikeevIlya** [reply]
> From my understanding, I don't believe there is a free EFI PXE version of memtest86+ anymore. Only the Pro version contains PXE support unfortunately:
> 
> https://www.memtest86.com/compare.html

Hello. How about adding memtest86plus to netboot.xyz? I see it on the official website https://memtest.org / images .iso memtest86plus utilities

**Reply by cros13** [reply]
It would be nice to have support for both Memtest86 Pro (i have licensing) and Memtest+.

**Comment by antonym** [comment]
I have found that you can `sanboot` the extracted USB image (https://www.memtest86.com/downloads/memtest86-usb.zip) and it works fine in EFI from iPXE. I have not put it up on the menu yet as it needs to be extracted and hosted on github and there appears to be a sanboot issue with loading a file from github due to the https redirections.

**Reply by x86dev** [reply]
Thanks for the hint -- would be nice if you put up an update here once you found a solution for this.

**Comment by UntouchedWagons** [comment]
How about https://www.memtest.org/ it can be booted via PXE

**Comment by antonym** [comment, marked as answer]
Support was added a few weeks ago to netbook.xyz so you can use memtest+ on UEFI now.

**Reply by cryptonym64** [reply]
memtest86+ 8.0 on 3.03 netbootxyz.  As soon as I select it, it clears screen and shows url to download. Then keyboard instantly shuts off. I had to switch the PXE settings on router to server the ...-legacy variants of the netbootxyz loaders as it was disabling keyboard the same way as it loaded.
